pkg/chartutil/util.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package chartutil
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// chartName is a regular expression for testing the supplied name of a chart.
+// This regular expression is probably stricter than it needs to be. We can relax it
+// somewhat. Newline characters, as well as $, quotes, +, parens, and % are known to be
+// problematic.
+var chartName = regexp.MustCompile("^[a-zA-Z0-9._-]+$")
+
+// maxChartNameLength is lower than the limits we know of with certain file systems,
+// and with certain Kubernetes fields.
+const maxChartNameLength = 250
+
+// ValidateChartName validate chartName
+func ValidateChartName(name string) error {
```

Comment:
Why was this moved from chartutil/create.go? "chartutil/util" includes "util" twice in the name. Can you please move this back?

Reply:
>Why was this moved from chartutil/create.go? "chartutil/util.go" includes "util" twice in the name. Can you please move this back?

Because both `create.go` and `chartfile.go` have the same logic for verifying chartname, I extracted this code in one place

> Alternatively, rename it as validation.go.

There seems to be a `validate_name.go` file before, with `ValidateReleaseName` and `ValidateMetadataName` inside, I put `ValidateChartName` here